xarray/core/dataset.py

```diff
@@ -1475,14 +1475,14 @@ def __getitem__(self, key):
         else:
             return self._copy_listed(np.asarray(key))
 
-    def __setitem__(self, key: Hashable, value) -> None:
-        """Add an array to this dataset.
+    def __setitem__(self, key, value) -> None:
```

Comment:
Can we add to the typing rather than removing it? Let me know if you want a hand

Reply:
Ok this should look better now :)

Reply:
Great. I _think_ this could have been a Union, but it's completely fine like this too.

Reply:
Tried this and `mypy` complains that `Single overload definition, multiple required`, it seems related to this https://github.com/python/mypy/issues/5047